packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -34,6 +34,179 @@ export interface HttpRouteMatch {
    * and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
    */
   readonly prefixPath: string;
+
+  /**
+   * Specifies the client request headers to match on
+   * @default - do not match on headers
+   */
+  readonly headers?: HttpRouteHeader[];
+
+  /**
+   * The client request method to match on. Specify only one.
+   * @default - do not match on request method
+   */
+  readonly method?: HttpRouteMatchMethod;
+
+  /**
+   * The client request scheme to match on. Specify only one. Applicable only for HTTP2 routes.
+   * @default - do not match on HTTP2 scheme
+   */
+  readonly scheme?: HttpRouteMatchScheme;
+}
+
+/**
+ * Supported request matching methods.
+ */
+export enum HttpRouteMatchMethod {
+  /**
+   * GET request
+   */
+  GET = 'GET',
+
+  /**
+   * HEAD request
+   */
+  HEAD = 'HEAD',
+
+  /**
+   * POST request
+   */
+  POST = 'POST',
+
+  /**
+   * PUT request
+   */
+  PUT = 'PUT',
+
+  /**
+   * DELETE request
+   */
+  DELETE = 'DELETE',
+
+  /**
+   * CONNECT request
+   */
+  CONNECT = 'CONNECT',
+
+  /**
+   * OPTIONS request
+   */
+  OPTIONS = 'OPTIONS',
+
+  /**
+   * TRACE request
+   */
+  TRACE = 'TRACE',
+
+  /**
+   * PATCH request
+   */
+  PATCH = 'PATCH',
+}
+
+/**
+ * Supported :scheme options for HTTP2
+ */
+export enum HttpRouteMatchScheme {
+  /**
+   * Match HTTP requests
+   */
+  HTTP = 'http',
+
+  /**
+   * Match HTTPS requests
+   */
+  HTTPS = 'https',
+}
+
+/**
+ * An object that represents the HTTP header in the request to match.
+ */
+export interface HttpRouteHeader {
+  /**
+   * Specify `true` to match anything except the match criteria.
+   * @default false
+   */
+  readonly invert?: boolean;
+
+  /**
+   * The method to use to match the header.
+   * @default - no match method
+   */
+  readonly match?: IHeaderMatchMethod;
+
+  /**
+   * A name for the HTTP header in the client request that will be matched on.
+   * @example 'content-type'
+   */
+  readonly name: string;
+}
+
+/**
+ * Match method.
+ */
+export interface IHeaderMatchMethod {
+  /**
+   * Produce the match method.
+   */
+  renderMatchMethod(): CfnRoute.HeaderMatchMethodProperty;
+}
+
+/**
+ * Used to generate header matching methods.
+ */
+export abstract class HeaderMatchMethod {
```

Comment:
Let's use the `bind()` pattern here instead.

1. Rename `IHeaderMatchMethod` to `HeaderMatchMethodConfig`, and make it have a `readonly matchProperty:  CfnRoute.HeaderMatchMethodProperty;` as its only property.
2. Create an abstract `bind(scope: Construct): HeaderMatchMethodConfig` method in `HeaderMatchMethod`.
3. Create a module-private implementation class that extends `HeaderMatchMethod` that is returned from each of the static factory methods in `HeaderMatchMethod` (whose return type should each be `HeaderMatchMethod`).
4. Call `bind()` in the implementation of  each `RouteSpec`.

The AppMesh module uses this pattern quite often, you can look at how it's implemented in other places for further guidance (for example, [for `ClientPolicy`](https://github.com/aws/aws-cdk/blob/ba99ad086e309fe5d11c75351e4b5dbb1f199478/packages/%40aws-cdk/aws-appmesh/lib/client-policy.ts#L8-L113)).